src/Model/Photo.php

```diff
@@ -1017,16 +1038,129 @@ public static function upload(int $uid, array $files)
 
 		$picture = [];
 
-		$picture['id']        = $photo['id'];
-		$picture['size']      = $photo['datasize'];
-		$picture['width']     = $photo['width'];
-		$picture['height']    = $photo['height'];
-		$picture['type']      = $photo['type'];
-		$picture['albumpage'] = DI::baseUrl() . '/photos/' . $user['nickname'] . '/image/' . $resource_id;
-		$picture['picture']   = DI::baseUrl() . '/photo/{$resource_id}-0.' . $Image->getExt();
-		$picture['preview']   = DI::baseUrl() . '/photo/{$resource_id}-{$smallest}.' . $Image->getExt();
+		$picture['id']          = $photo['id'];
+		$picture['resource_id'] = $resource_id;
+		$picture['size']        = $photo['datasize'];
+		$picture['width']       = $photo['width'];
+		$picture['height']      = $photo['height'];
+		$picture['type']        = $photo['type'];
+		$picture['albumpage']   = DI::baseUrl() . '/photos/' . $user['nickname'] . '/image/' . $resource_id;
+		$picture['picture']     = DI::baseUrl() . '/photo/{$resource_id}-0.' . $Image->getExt();
+		$picture['preview']     = DI::baseUrl() . '/photo/{$resource_id}-{$smallest}.' . $Image->getExt();
+
+		$Image->__destruct();
 
 		Logger::info('upload done', ['picture' => $picture]);
 		return $picture;
 	}
+
+	/**
+	 *
+	 * @param int   $uid   User ID
+	 * @param array $files uploaded file array
+	 * @return string avatar resource
+	 */
+	public static function uploadAvatar(int $uid, array $files)
+	{
+		$data = self::uploadImage($files);
+		if (empty($data)) {
+			return [];
```

Comment:
Because the return type is string too:
```suggestion
			return '';
```